src/libraries/System.Runtime.Serialization.Xml/ref/System.Runtime.Serialization.Xml.cs

```diff
@@ -6,6 +6,50 @@
+        public string? ContractType { get { throw null; } }
+        internal DataContract(DataContractCriticalHelper helper) { }
+        internal const System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes DataContractPreserveMemberTypes =
+            System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicMethods |
+            System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.NonPublicMethods |
+            System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicConstructors |
+            System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.NonPublicConstructors |
+            System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicFields |
+            System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicProperties;
+        public static string EncodeLocalName(string localName) { throw null; }
+        public GenericInfo? GenericInfo { [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("Data Contract Serialization and Deserialization might require types that cannot be statically analyzed. Make sure all of the required types are preserved.")] get { throw null; } }
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("Data Contract Serialization and Deserialization might require types that cannot be statically analyzed. Make sure all of the required types are preserved.")]
+        public System.Xml.XmlQualifiedName GetArrayTypeName(bool isNullable) { throw null; }
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("Data Contract Serialization and Deserialization might require types that cannot be statically analyzed. Make sure all of the required types are preserved.")]
+        public static DataContract? GetBuiltInDataContract(string name, string ns) { throw null; }
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("Data Contract Serialization and Deserialization might require types that cannot be statically analyzed. Make sure all of the required types are preserved.")]
+        public static DataContract GetDataContract(Type type) { throw null; }
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("Data Contract Serialization and Deserialization might require types that cannot be statically analyzed. Make sure all of the required types are preserved.")]
+        public static System.Xml.XmlQualifiedName GetStableName(Type type) { throw null; }
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("Data Contract Serialization and Deserialization might require types that cannot be statically analyzed. Make sure all of the required types are preserved.")]
+        public static Type GetSurrogateType(ISerializationSurrogateProvider surrogateProvider, Type type) { throw null; }
+        public bool HasRoot { get { throw null; } }
```

Comment:
It looks like this doesn't need any internal implementation which won't already be exposed. Can the logic of this method be duplicated instead?

Reply:
Similar to EncodeLocalName... both surrogate provider interfaces are defined in the framework, and the DCS inner workings already has a pattern for managing/executing surrogate providers. GetSurrogateType() and DataContractSet.TryGetSurrogateData() are two APIs here which we can do in an external package - but surrogate handling feels like a DC responsibility, so I have these two methods to keep responsibility with DC/DCSet.

But as with EncodeLocalName, if there is pushback in the API review, these two can be taken out and done in the schema package if needed.